Source/WebKit/Scripts/generate-serializers.py

```diff
@@ -610,6 +610,9 @@ def generate_impl(serialized_types, serialized_enums, headers):
                 if valid_value.condition is not None:
                     result.append('#if ' + valid_value.condition)
                 result.append('        ' + ('' if i == 0 else '| ') + 'static_cast<' + enum.underlying_type + '>(' + enum.namespace_and_name() + '::' + valid_value.name + ')' + (';' if i == len(enum.valid_values) - 1 else ''))
+                if valid_value.condition is not None and i == len(enum.valid_values) - 1:
```

Comment:
Changes to generated-serializers.py need to be tested by making Source/WebKit/Scripts/webkit/tests/TestSerializedType.serialization.in hit the change and running "make -C Source/WebKit/Scripts/webkit/tests" to update the output.

I don't think this is a good change.  I think instead we should make the generated output look like this:

    constexpr uint8_t allValidBitsValue = 0
        | static_cast<uint8_t>(EnumNamespace2::OptionSetEnumType::OptionSetFirstValue)
#if ENABLE(OPTION_SET_SECOND_VALUE)
        | static_cast<uint8_t>(EnumNamespace2::OptionSetEnumType::OptionSetSecondValue)
#endif
        | static_cast<uint8_t>(EnumNamespace2::OptionSetEnumType::OptionSetThirdValue)
        | 0;

That'll be robust against the first, last, or all enum values having a condition.